doc_src/index.hdr.in

```diff
@@ -17,7 +17,201 @@
+- <a href="#shell-syntax">Command Syntax</a>: Shell commands have a standard syntax
+- <a href="#shell-programming">Commands versus Programs</a>: Commands differ from normal programs
+- <a href="#shebang-line">Shebang Line</a>: How the shell knows the language of a script
+
+\subsection shell-standards Shell Standards
+
+A shell is an interface to the operating system that reads from the commandline of a terminal. A shell's task is to identify and interpret commands. The commands can come from different applications and can be written in different programming languages.
+
+This can only work smoothly if shells adapt to some common standards. For shells there is the POSIX standard, see <a href="https://en.wikipedia.org/wiki/Command-line_interface#Command-line_interpreter">Command-line interpreters</a>. `fish` tries to satisfy the POSIX standard wherever it does not get into the way of its own design principles, see <a href="design.html">`fish` Design</a>.
+
+\subsection man-page Manual Pages
+
+There is a common standard on how to receive help on shell commands: applications provide a manual page to their commands that can be opened with the `man` command:
+
+\fish
+>man COMMAND
+\endfish
+
+This convention helps to make sure help can be found on commands no matter where they originate from. *fish*'s internal commands all come with a manual page.
+
+\subsection shell-syntax Command Syntax
+
+Shell commands also have some syntax in common, no matter where they originate from: 
+
+Commands consist of three parts:
+
+\fish
+COMMAND [OPTIONS] ARGUMENTS
+\endfish
+
+- <b>`COMMAND`</b>: the name of the executeable
+
+- <b>`[OPTIONS]`</b>: options can change the behaviour of a command
+
+- <b>`ARGUMENTS`</b>: input to the command
+
+For external <b>commands</b> the executable is usually stored on file and the file path must be included in the variable `$PATH`, so that the file can be found: see <a href="#variables-special">Special Variables</a>.
+
+<b>Options</b> come in two forms: a short name, that is a hyphen with a single letter; or a long name, consisting of two hyphens with words connected by hyphens. Example: `-h` and `--help` are common options for opening the manual page of a command. Options are a fixed set, described in the manual pages of the command, see <a href="#man-page">Manual Pages</a>
+
+<b>Arguments</b> are the arbitrary input part of a command: often it is a file or directory name, sometimes it is a string or a list.
+
+Example:
+
+\fish
+>echo -s Hallo World!
+HalloWorld!
+\endfish
+
+-  both `Hallo` and `World!` are arguments to the echo command
+- `-s` is an option that suppresses spaces in the output of the command
+
+\subsection shell-programming Commands versus Programs
+
+Shell commands differ from other programs in the following way:
+
+- they cannot return variables
+
+Instead:
+
+- They transform an <b>input stream</b> into an <b>output stream</b>. Both of these streams are usually the terminal, but they can be redirected.
```

Comment:
This seems a bit math-y to me? "They take an input stream and write to an output stream"?

Reply:
I took your suggestion and changed my text.